src/ssz/type/bit_list.zig

```diff
@@ -79,6 +79,46 @@ pub fn BitList(comptime limit: comptime_int) type {
             return true_bit_count;
         }
 
+        pub fn getSingleTrueBit(self: *const @This(), out: *?usize) !void {
```

Comment:
two things:
- no need for error union here and below
- in this case, where out is a single `?usize` we don't need to update the return value via pointer. Can/should just return the `?usize`. Going back to first principles, afaik the reason we pass in an `out` pointer is to avoid allocating a big something inside the function or accidentally copying a big something. Usually a slice or array. In the case of a slice, it means we need an allocator inside the function. In the case of an array, it means we may copy a big array in some circumstances.